components/eam/src/chemistry/modal_aero/modal_aero_amicphys.F90

```diff
@@ -4889,57 +4890,113 @@ subroutine mam_pcarbon_aging_1subarea(                        &
       fac_volsfc = exp( 2.5*(alnsg_aer(nfrm)**2) )
       xferfrac_max = 1.0_r8 - 10.0_r8*epsilon(1.0_r8)   ! 1-eps
 
-      tmp1 = vol_shell*dgn_a(nfrm)*fac_volsfc
-      tmp2 = max( 6.0_r8*dr_so4_monolayers_pcage*vol_core, 0.0_r8 )
-      if (tmp1 >= tmp2) then
+      xferfrac_tmp1 = vol_shell*dgn_a(nfrm)*fac_volsfc
+      xferfrac_tmp2 = max( 6.0_r8*dr_so4_monolayers_pcage*vol_core, 0.0_r8 )
+      if (xferfrac_tmp1 >= xferfrac_tmp2) then
          xferfrac_pcage = xferfrac_max
       else
-         xferfrac_pcage = min( tmp1/tmp2, xferfrac_max )
-      end if
+         xferfrac_pcage = min( xferfrac_tmp1/xferfrac_tmp2, xferfrac_max )
+      endif
 
-      do iaer = 1, naer
-         if (lmap_aer(iaer,nfrm) > 0) then
-            ! species is pom or bc
-            ! transfer the aged fraction to accum mode
-            ! include this transfer change in the cond and/or coag change (for mass budget)
-            tmpa = qaer_cur(iaer,nfrm)*xferfrac_pcage
-            qaer_cur(iaer,nfrm)      = qaer_cur(iaer,nfrm) - tmpa
-            qaer_cur(iaer,ntoo)      = qaer_cur(iaer,ntoo) + tmpa
-            qaer_del_cond(iaer,nfrm) = qaer_del_cond(iaer,nfrm) - tmpa*tmp3
-            qaer_del_cond(iaer,ntoo) = qaer_del_cond(iaer,ntoo) + tmpa*tmp3
-            qaer_del_coag(iaer,nfrm) = qaer_del_coag(iaer,nfrm) - tmpa*tmp4
-            qaer_del_coag(iaer,ntoo) = qaer_del_coag(iaer,ntoo) + tmpa*tmp4
-         else
-            ! species is soa, so4, or nh4 produced by condensation or coagulation
-            ! transfer all of it to accum mode
-            ! also transfer the condensation and coagulation changes
-            !    to accum mode (for mass budget)
-            qaer_cur(iaer,ntoo)      = qaer_cur(iaer,ntoo) &
-                                     + qaer_cur(iaer,nfrm)
-            qaer_del_cond(iaer,ntoo) = qaer_del_cond(iaer,ntoo) &
-                                     + qaer_del_cond(iaer,nfrm)
-            qaer_del_coag(iaer,ntoo) = qaer_del_coag(iaer,ntoo) &
-                                     + qaer_del_coag(iaer,nfrm)
-            qaer_cur(iaer,nfrm)      = 0.0_r8
-            qaer_del_cond(iaer,nfrm) = 0.0_r8
-            qaer_del_coag(iaer,nfrm) = 0.0_r8
-         end if
-      end do
-      ! number - transfer the aged fraction to accum mode
-      ! include this transfer change in the cond and/or coag change (for mass budget)
-      tmpa = qnum_cur(nfrm)*xferfrac_pcage
-      qnum_cur(nfrm)      = qnum_cur(nfrm) - tmpa
-      qnum_cur(ntoo)      = qnum_cur(ntoo) + tmpa
-      qnum_del_cond(nfrm) = qnum_del_cond(nfrm) - tmpa*tmp3
-      qnum_del_cond(ntoo) = qnum_del_cond(ntoo) + tmpa*tmp3
-      qnum_del_coag(nfrm) = qnum_del_coag(nfrm) - tmpa*tmp4
-      qnum_del_coag(ntoo) = qnum_del_coag(ntoo) + tmpa*tmp4
+      return
+      end subroutine mam_pcarbon_aging_frac 
 
-      end do agepair_loop1
+      
+      subroutine transfer_aged_pcarbon_mass_to_accum(iaer, nfrm, ntoo, &
```

Comment:
`transfer_aged_pcarbon_mass_to_accum` and `transfer_aged_pcarbon_num_to_accum` can be combined into one subroutine. We can send different arguments (mass vs. number) to these. `transfer_aged_pcarbon_mass_to_accum` subroutine uses 2D array for the aerosol mass. We can do something like this to accommodate it:
```
call transfer_aged_pcarbon_mass_to_accum( nfrm, ntoo, &
xferfrac_pcage, frac_cond, frac_coag, &
qaer_cur(iaer,:), qaer_del_cond(iaer,:), qaer_del_coag(iaer,:))
```

Reply:
combined the two subroutine into one